src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -162,14 +163,14 @@ public ReplicateSupplyService(ReplicatesService replicatesService,
                 continue;
             }
 
-            final List<Replicate> replicates = replicatesService.getReplicates(chainTaskId);
             final boolean taskNeedsMoreContributions = ConsensusHelper.doesTaskNeedMoreContributionsForConsensus(
                     chainTaskId,
-                    replicates,
+                    replicatesList.getReplicates(),
                     task.getTrust(),
                     task.getMaxExecutionTime());
 
             if (taskNeedsMoreContributions) {
+                boolean isTeeTask = task.isTeeTask();
                 String enclaveChallenge = smsService.getEnclaveChallenge(chainTaskId, isTeeTask);
```

Comment:
`smsService.getEnclaveChallenge(chainTaskId, isTeeTask)` can be:
1. Totally moved to `TaskUpdateManager.received2Initializing` and stored in `task.enclaveChallenge` for later use
2. (In order to honor your cache) Called a first time in `TaskUpdateManager.received2Initializing`, to relocate real HTTP call outside this method.

Reply:
Excellent!